runtime/cudaq/platform/fermioniq/helpers/FermioniqServerHelper.cpp

```diff
@@ -0,0 +1,512 @@
+}
+
+// Helper function to get a value from a dictionary or return a default
+template <typename K, typename V>
+V getOrDefault(const std::unordered_map<K, V> &map, const K &key,
+               const V &defaultValue) {
+  auto it = map.find(key);
+  return (it != map.end()) ? it->second : defaultValue;
+}
+
+// Check if a key exists in the backend configuration
+bool FermioniqServerHelper::keyExists(const std::string &key) const {
+  return backendConfig.find(key) != backendConfig.end();
+}
+
+ServerJobPayload
+FermioniqServerHelper::createJob(std::vector<KernelExecution> &circuitCodes) {
+  cudaq::debug("createJob");
+
+  if (circuitCodes.size() != 1) {
+    throw std::runtime_error("Fermioniq allows only one circuit codes.");
+  }
+
+  auto job = nlohmann::json::object();
+  auto circuits = nlohmann::json::array({"__qir_base_compressed__"});
+  auto configs = nlohmann::json::array();
+  auto noise_models = nlohmann::json::array();
+
+  std::vector<std::string> circuit_names;
+
+  for (auto &circuitCode : circuitCodes) {
+    cudaq::info("name: {}", circuitCode.name);
+
+    circuit_names.push_back(circuitCode.name);
+
+    cudaq::info("outputNames: {}", circuitCode.output_names.dump());
+
+    // Construct the job message (for Fermioniq backend)
+    circuits.push_back(circuitCode.code);
+
+    auto config = nlohmann::json::object();
+    config["n_shots"] = static_cast<int>(shots);
+    if (keyExists(CFG_BOND_DIM_KEY)) {
+      config["bond_dim"] = stoi(backendConfig.at(CFG_BOND_DIM_KEY));
+    }
+
+    if (circuitCode.user_data.contains("observable")) {
+      config["observable"] = circuitCode.user_data["observable"];
+    }
+
+    configs.push_back(config);
+
+    // To-DO: Add noise models
+    noise_models.push_back(nullptr);
+  }
+
+  auto circuit_names_imploded = std::accumulate(
+      circuit_names.begin(), circuit_names.end(), std::string(),
+      [](const std::string &a, const std::string &b) -> std::string {
+        return a + (a.length() > 0 ? "," : "") + b;
+      });
+
+  if (keyExists(CFG_REMOTE_CONFIG_KEY)) {
+    job["remote_config"] = backendConfig.at(CFG_REMOTE_CONFIG_KEY);
+  }
+  job["circuit"] = circuits;
+  job["config"] = configs;
+  job["noise_model"] = noise_models;
+  job["verbosity_level"] = 1;
+  job["label"] = circuit_names_imploded;
+  if (keyExists(CFG_PROJECT_ID_KEY)) {
+    job["project_id"] = backendConfig.at(CFG_PROJECT_ID_KEY);
+  }
+
+  auto payload = nlohmann::json::array();
+  payload.push_back(job);
+
+  // Return a tuple containing the job path, headers, and the job message
+  auto job_path = backendConfig.at(CFG_URL_KEY) + "/api/jobs";
+  auto ret = std::make_tuple(job_path, getHeaders(), payload);
+  return ret;
+}
+
+/// Refresh the api key and refresh-token
+void FermioniqServerHelper::refreshTokens(bool force_refresh) {
+  std::mutex m;
+  std::lock_guard<std::mutex> l(m);
+  RestClient client;
+
+  if (!force_refresh) {
+    auto now = std::chrono::high_resolution_clock::now();
```

Comment:
Is there a reason we use `std::chrono::high_resolution_clock` here but `std::chrono::system_clock` for the `tokenExpTime`.
This caused the compile error with Clang (the installer CI).

Reply:
thanks. I indeed saw that and will update it.